src/main/java/com/google/sps/data/Article.java

```diff
@@ -0,0 +1,29 @@
+package com.google.sps.data;
+
+public class Article {
+  private String title;
+  private String link;
+
+  private Article(Builder b) {
+    this.title = b.title;
+    this.link = b.link;
+  }
+
+  public static class Builder {
+    private String title;
+    private String link;
+
+    public Builder(String title) {
+      this.title = title;
```

Comment:
I think you are missing a "return this;" in this function.  That will cause the method chaining to break.

Reply:
I suppose this does depend on if you want "title" to be the only required parameter and the link parameter to be optional, in which case this design is ok.

Reply:
I'm a little confused about this comment -- should I put "return this;" in the builder constructor?

Reply:
Sorry, my suggestion accurate in terms of what I was think of so let me correct myself.

First of all, no, you don't need a 'return this' in the constructor.  However, in the constructor you only include parameters that are required.  Then, you want a 'with' function for each optional parameter, or for parameters that can be set at a later time.

I was considering whether you needed to include title and link as part of the constructor, or whether you wanted a separate 'withTitle' and 'withLink' function.  On second look, I think this design is fine.  You can optionally move link to it's own 'withLink' function but either way is reasonable.